lib/DDGC/Web/Controller/Ideas.pm

```diff
@@ -186,6 +186,26 @@ sub idea : Chained('idea_id') PathPart('') Args(1) {
 	$c->stash->{title} = $c->stash->{idea}->title;
 }
 
+sub claim : Chained('idea_id') Args(0) {
+	my ( $self, $c ) = @_;
+	$c->require_action_token;
+	return $c->detach if (!$c->user);
+
+	if ( !$c->stash->{idea}->claimed_by ) {
+		$c->stash->{idea}->update( {
+			claimed_by => $c->user->id,
+		} );
+	}
+	elsif ( $c->stash->{idea}->claimed_by &&
+	     ( $c->stash->{idea}->claimed_by == $c->user->id ||
+	       $c->user->is('admin') ) ) {
+		$c->stash->{idea}->update( {
+			claimed_by => undef,
+		} );
+	}
+	$c->response->redirect( $c->chained_uri(@{ $c->stash->{idea}->u }) );
```

Comment:
I guess the question here is: after we create an IA Page for this idea (and I think the claim button shouldn't exist until that happens) and a user claims it, should we update the developer field for that IA Page as well? I think we should.
Even though, if the user ends up doing nothing it will be a little bit of a pain to remove him/her from the developer array...